src/System.Runtime.Extensions/tests/System/Environment.MachineName.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Tests
+{
+    public class Environment_MachineName
+    {
+        // [Fact] Skip test for the mscolib changes to reach corefx
+        public void TestMachineNameProperty()
+        {
+            string computerName = GetComputerName();
+            Assert.Equal(computerName, Environment.MachineName);
+        }
+
+        [DllImport("kernel32.dll", EntryPoint = "GetComputerNameW")]
+        private unsafe static extern int GetComputerName(char* nameBuffer, ref int bufferSize);
+
+        [DllImport("libc")]
+        private unsafe static extern int gethostname(char* name, int len);
+
+        private unsafe static string GetComputerName()
+        {
+            const int MaxMachineNameLength = 256;
```

Comment:
The code for the wrapper is exactly same as what you have in the test. If there is a bug in the wrapper that causes wrong value to be returned, nobody will notice - because the test will have the exact same bug.

Good tests get the value to compare against in a different way than a product code that they are testing.

> Environment.SetEnvironmentVariable("COMPUTERNAME", "myprogramvalue");

It is not ok for other tests running in the same process to corrupt global settings like this. There is no point in hardening tests against corrupted global settings.